### PR TITLE
update sparql

### PR DESCRIPTION
**Pull request by Treesarj**
# Description:
This updates the feature that fetch the relations from KG when graph uri is provided in the mapping strategy.
Updates are:
-  The SPARQL query is modified to fetch all the properties with the required predicates of the selected sub graph
-  The SPARQL query was modified to include the full graph if no common parent class was found 

## Type of change:

- [ ] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand, including clearly named variables?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: oteapi_dlite/strategies/mapping.py [hunk 1]
```diff
@@ -242,15 +242,27 @@ class URIs within a specified RDF graph.
...
+                FILTER(
+                    {% for class_name in class_names -%}
+                        ?class = <{{ class_name }}>{{ " ||" if not loop.last }}
+                    {% endfor %}
+                )
+                }
+            UNION
+                {
+                # Explicitly include owl:Thing as a parent class
+                VALUES ?class { {% for class_name in class_names -%} <{{ class_name }}> {% endfor %} }
```
Comment (quaat) [inline oteapi_dlite/strategies/mapping.py]: I don't see how this fixes anything. You set `Thing` as parent for all parentClasses?!. I would rather suggest that you leave the original query, but in the code that calls it, if the query doesn't return a single parent class, call a different method that fetches the entire graph.

File: oteapi_dlite/strategies/mapping.py [hunk 2]
```diff
@@ -326,26 +338,46 @@ def fetch_and_populate_graph(
...
             ?subject ?predicate ?object .
-            ?subject rdfs:subClassOf* <{parent_node}> .
+            }}
+
+            # Ensure subject, predicate, and object are not empty
+            FILTER(BOUND(?subject) && BOUND(?predicate) && BOUND(?object))
+
+            # Filter by the specific predicates
             FILTER (?predicate IN (
-                rdfs:subClassOf,
```
Comment (quaat) [inline oteapi_dlite/strategies/mapping.py:336]: I think we need the `subClassOf` and perhaps even add `hasSubClass`. Sometime we will need to traverse the graph up and down the inheritance levels.

**Review by quaat — CHANGES_REQUESTED** [review]

File: oteapi_dlite/strategies/mapping.py (hunk 2 above)
Comment (Treesarj) [reply oteapi_dlite/strategies/mapping.py:336]: ok will add it back

File: oteapi_dlite/strategies/mapping.py (hunk 1 above)
Comment (Treesarj) [reply oteapi_dlite/strategies/mapping.py]: > I don't see how this fixes anything. You set `Thing` as parent for all parentClasses?!. I would rather suggest that you leave the original query, **but in the code that calls it, if the query doesn't return a single parent class, call a different method that fetches the entire graph.**

separate method sounds cleaner

**Review by quaat — APPROVED** [review]

**Pull request merged**